streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -1015,4 +1015,55 @@ public void shouldThrowPatternNotValidForTopicNameException() {
             assertEquals(str, exception.getMessage());
         }
     }
+
+    @Test
+    public void shouldProcessStreamProcessingGuaranteeAsExactlyOnce() {
```

Comment:
I think, this test can be simplified. We only want to make sure, that EOS works.

As an alternative, we should run each test with and w/o EOS -- that way, we don't need to duplicate any code.

Reply:
I agree with @mjsax about running existing tests.  

To that end, we can use JUnit parameters so that we can execute each existing test with EOS enabled and disabled.  `AbstractJoinIntegrationTest` can be used as an example of using JUnit parameters in a test.

Reply:
+1

Reply:
Fixed